model/database/orm.py

```diff
@@ -0,0 +1,100 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+A class responsible for creating a light weight django instance to
+access the Django ORM that defines the autoreduction database
+"""
+import os
+import sys
+import logging
+
+import django
+from django.conf import settings
+
+from utils.project.structure import get_project_root
+
+
+class DjangoORM:
+    """
+    Encapsulate Django setup and ORM access in a class
+    This should normally be used like the following:
+    orm = DjangoORM
+    orm.connect()
+    records = orm.data_model.<TableName>.objects.<Filter/OrderBy/..>
+    """
+
+    data_model = None
+    variable_model = None
+
+    @staticmethod
+    def add_webapp_path():
+        """
+        Get path to the autoreduce_webapp and add this path to the system path
+        This will enable importing of the autoreduce_webapp (and it's models)
+        """
+        path = os.path.join(get_project_root(), 'WebApp', 'autoreduce_webapp')
+        if path not in sys.path:
+            # Can't properly replicate this in a testing environment due to travis
+            # constraint on sys path access and mocking would break underlying dependency
+            # but it's simple enough we don't need to worry about testing it
+            sys.path.append(path)  # pragma: no cover
+
+    @staticmethod
+    def setup_django():
+        """
+        Use the WebApp settings to initialise a django instance that can be used for model access
+        """
+        try:
+            # import here to avoid failure without calling add_webapp_path first
+            # pylint:disable=import-outside-toplevel
```

Comment:
I wonder if these pylint disables which span across multiple levels ought to be applied at a class level?

Reply:
Probably not in this case. I generally take an approach depending on the severity of the disable. In this case (toplevel imports) is quite serious and as such I want to ensure that we are only disabling this at the point of use. 

Other things such as line too long and docstrings are less important so can be used at class level